weaver/utils.py

```diff
@@ -2872,6 +2872,14 @@ def fetch_directory(location,                           # type: str
                                     include=include, exclude=exclude,
                                     settings=settings, **option_kwargs)
     elif location.startswith("http://") or location.startswith("https://"):
+        # Next two lines are added to match behavior of `download_files_s3` and replicate input directory name
+        # in output location
+        input_directory_name = os.path.basename(os.path.normpath(location))
+        out_dir = os.path.join(out_dir, input_directory_name)
+        # This is to filter out request parameters from the input location and prevent them
+        # from being used as a directory
+        if "?" in out_dir:
+            out_dir = out_dir.split("?")[0]
```

Comment:
In which situation did you encounter this issue. I would expect the calling function to pass only the URL part with queries and anchor fragments already removed, as the same logic must apply of all protocols.

Reply:
It happened in this test https://github.com/crim-ca/weaver/blob/master/tests/test_utils.py#L643

It downloaded in a newly created folder named `?f=json`.

Since the function `fetch_directory` uses the location argument (https://github.com/crim-ca/weaver/blob/2c888096d318ecd3c6d67f2093de269c67c80afd/weaver/utils.py#L2877-L2878), I expect the filtering that removes the query and other fragments would be in another layer of the parsing, or somehow didn't work in the context of this test?

Reply:
I see. The value obtained here:
https://github.com/crim-ca/weaver/blob/2c888096d318ecd3c6d67f2093de269c67c80afd/weaver/utils.py#L2860-L2861 could be stored in a variable and used for resolving only the directory name.
It's important to keep the query arguments though for the following calls that could need it:
https://github.com/crim-ca/weaver/blob/2c888096d318ecd3c6d67f2093de269c67c80afd/weaver/utils.py#L2865-L2866
https://github.com/crim-ca/weaver/blob/2c888096d318ecd3c6d67f2093de269c67c80afd/weaver/utils.py#L2869
https://github.com/crim-ca/weaver/blob/2c888096d318ecd3c6d67f2093de269c67c80afd/weaver/utils.py#L2884

Reply:
You could do `loc_path = get_url_without_query(location)`, and then use `loc_path` in the current if. 
In the HTTP part, you can then work with `loc_part` directly in the code that resolves `out_dir`.